### DISCUSSION TITLE
setting WIFI SSID launches empty browser?

### DISCUSSION THREAD

**Discussion by eric-s-myers** [ClassicDIY/ClassicMQTT#41] [General]
first off... thank you!  I'm excited to get this going.  after working through my USB-UART drivers, I got my esp flashed.  loaded.  set the Wifi SSID/pwd and then save... but an empty browser launches.  okay.  what the heck is the default IP for the site!?  Can't find it anywhere.

**Comment by eric-s-myers** [comment]
I tried http://192.168.86.219/ based on the screenshot in the wiki, but it's not loading.  http connection timeout and can't ping that IP.

**Comment by eric-s-myers** [comment]
@graham22 ^ any input here would be appreciated!

**Comment by eric-s-myers** [comment]
here's the output from flashing... so not looking like an IP is getting set... I see an empty string....

{"IP":"","ApPassword":"ClassicMQTT"}
Device IP: http://
******* Use admin/ClassicMQTT to enter configuration page *******

**Comment by eric-s-myers** [comment]
I connected with Arduino IDE and wrote out the IP address the board was assigned by the router.  that has seemed to resolve it.   I was able to get to the config page at that IP.

**Comment by graham22** [comment]
Uploaded a new version (1.3.5) that has a fix to publish the ip address to the flasher tool

**Comment by eric-s-myers** [comment]
thanks!  yes, that resolved it and confirms that is the correctly assigned IP.  I have it all up and running... but I don't get a ClassicMQTT access point nor am I seeing any messages in MQTT explorer in localhost.  I can access the esp32 on the network at the assigned IP.  I get the ClassicMQTT config screen, too.  It all appears correct.  I can confirm my MQTT server is right because I'm getting MQTT messages in MQTT-explorer when I publish data manually.  am I misunderstanding something?  ClassicMQTT will publish the SCC data to the server I have configured?  so, in this case, my local mosquitto instance.

**Comment by graham22** [comment]
Try sending a wake command

[image: image.png]

or stay awake

[image: image.png]


[image: image.png]


On Thu, Feb 1, 2024 at 8:20 AM eric-s-myers ***@***.***>
wrote:

> thanks! yes, that resolved it and confirms that is the correctly assigned
> IP. I have it all up and running... but I don't get a ClassicMQTT access
> point nor am I seeing any messages in MQTT explorer in localhost. I can
> access the esp32 on the network at the assigned IP. I get the ClassicMQTT
> config screen, too. It all appears correct. I can confirm my MQTT server is
> right because I'm getting MQTT messages in MQTT-explorer when I publish
> data manually. am I misunderstanding something? ClassicMQTT will publish
> the SCC data to the server I have configured? so, in this case, my local
> mosquitto instance.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/ClassicDIY/ClassicMQTT/discussions/41#discussioncomment-8334843>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABIYATMFGQLVNFP6RG5V5QTYROJDJAVCNFSM6AAAAABCTWFWY2VHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4DGMZUHA2DG>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
Graham A. Ross
(613) 769-1331

**Reply by eric-s-myers** [reply]
I'm not sure what 'wake' command means.  the attached image isn't uploaded.  I appreciate the help!

**Comment by graham22** [comment]
I'm using the MQTT.fx tool to send a cmd

![Capture](https://github.com/ClassicDIY/ClassicMQTT/assets/5341261/08d40393-5f08-4508-91d5-66aaca1b630e)

**Comment by eric-s-myers** [comment]
ah... so the problem is classicMQTT is actively refusing a connection.  "Connection refused: no further information: 192.168.0.236:1883"

telnet says it can ping the IP, but TCP connect via port 1883 fails

**Comment by graham22** [comment]
Just to make sure I understand, 192.168.0.236:1883 is the address of your MQTT broker
Can you connect to it using MQTT.fx or MQTT explorer?
and ClassicMQTT is reporting that it can't connect to your broker

**Reply by eric-s-myers** [reply]
yes, that's the correct address for ClassicMQTT.  no... neither MQTT.fx nor MQTT-explorer can connect to ClassicMQTT

**Comment by graham22** [comment]
wait a minute!, MQTT.fx connects to your MQTT broker on your local network, not to ClassicMQTT.
ClassicMQTT is an MQTT publisher, it will read Modbus data from your classic and publish it to a broker like Mosquitto.
Tools like MQTT.fx connect to the broker as a subscriber they don't connect directly to ClassicMQTT

**Reply by eric-s-myers** [reply]
right, ClassicMQTT is not publishing.  I've confirmed my mosquitto instance is not getting any events from ClassicMQTT.  I've also confirmed the broker isn't getting classic messages.  other messages are, so it's a ClassicMQTT issue.

**Comment by graham22** [comment]
Can you send me the logs
my esp logs look like the following;
![Capture](https://github.com/ClassicDIY/ClassicMQTT/assets/5341261/a4e1b66f-a415-4247-91c8-41bf3f32831a)

**Comment by eric-s-myers** [comment]
COMM errors galore... haha.

[  3319][E][main.cpp:319] modbusErrorCallback(): packetId[0x2], error[COMM_ERROR]
[  5315][E][main.cpp:319] modbusErrorCallback(): packetId[0x3], error[COMM_ERROR]
[  6291][I][main.cpp:457] connectToMqtt(): Connecting to MQTT...
[  6298][D][main.cpp:470] connectToMqtt(): _mqttRootTopic: classic
[  6299][D][main.cpp:471] connectToMqtt(): rootTopicPrefix: classic/c200
[  7310][E][main.cpp:319] modbusErrorCallback(): packetId[0x4], error[COMM_ERROR]
[  9308][E][main.cpp:319] modbusErrorCallback(): packetId[0x5], error[COMM_ERROR]
[ 11310][E][main.cpp:319] modbusErrorCallback(): packetId[0x6], error[COMM_ERROR]
[ 13316][E][main.cpp:319] modbusErrorCallback(): packetId[0x7], error[COMM_ERROR]
[ 15323][E][main.cpp:319] modbusErrorCallback(): packetId[0x8], error[COMM_ERROR]
[ 17316][E][main.cpp:319] modbusErrorCallback(): packetId[0x9], error[COMM_ERROR]
[ 19316][E][main.cpp:319] modbusErrorCallback(): packetId[0xa], error[COMM_ERROR]
[ 21320][E][main.cpp:319] modbusErrorCallback(): packetId[0xb], error[COMM_ERROR]
[ 23319][E][main.cpp:319] modbusErrorCallback(): packetId[0xc], error[COMM_ERROR]
[ 24822][I][main.cpp:444] onMqttDisconnect(): Disconnected from MQTT. Reason: 0
[ 25318][E][main.cpp:319] modbusErrorCallback(): packetId[0xd], error[COMM_ERROR]
[ 27322][E][main.cpp:319] modbusErrorCallback(): packetId[0xe], error[COMM_ERROR]
[ 29325][D][main.cpp:417] modbusCallback(): packetId[0xf], slaveAddress[0xa], functionCode


....



[125365][D][main.cpp:417] modbusCallback(): packetId[0x3f], slaveAddress[0xa], functionCode[0x3], numberOfRegisters[22]
[127364][D][main.cpp:417] modbusCallback(): packetId[0x40], slaveAddress[0xa], functionCode[0x3], numberOfRegisters[44]
[129365][D][main.cpp:417] modbusCallback(): packetId[0x41], slaveAddress[0xa], functionCode[0x3], numberOfRegisters[22]
[131366][D][main.cpp:417] modbusCallback(): packetId[0x42], slaveAddress[0xa], functionCode[0x3], numberOfRegisters[44]
[133369][D][main.cpp:417] modbusCallback(): packetId[0x43], slaveAddress[0xa], functionCode[0x3], numberOfRegisters[22]
[135368][D][main.cpp:417] modbusCallback(): packetId[0x44], slaveAddress[0xa], functionCode[0x3], numberOfRegisters[44]
[137372][D][main.cpp:417] modbusCallback(): packetId[0x45], slaveAddress[0xa], functionCode[0x3], numberOfRegisters[22]
[139373][D][main.cpp:417] modbusCallback(): packetId[0x46], slaveAddress[0xa], functionCode[0x3], numberOfRegisters[44]
[140637][I][main.cpp:444] onMqttDisconnect(): Disconnected from MQTT. Reason: 0
[141374][D][main.cpp:417] modbusCallback(): packetId[0x47], slaveAddress[0xa], functionCode[0x3], numberOfRegisters[22]
[143372][D][main.cpp:417] modbusCallback(): packetId[0x48], slaveAddress[0xa], functionCode[0x3], numberOfRegisters[44]
[145373][D][main.cpp:417] modbusCallback(): packetId[0x49], slaveAddress[0xa], functionCode[0x3], numberOfRegisters[22]
[145636][I][main.cpp:457] connectToMqtt(): Connecting to MQTT...
[145643][D][main.cpp:470] connectToMqtt(): _mqttRootTopic: classic
[145643][D][main.cpp:471] connectToMqtt(): rootTopicPrefix: classic/c200/c200/c200/c200/c200/c200/c200
[147376][D][main.cpp:417] modbusCallback(): packetId[0x4a], slaveAddress[0xa], functionCode[0x3], numberOfRegisters[44]
[149379][D][main.cpp:417] modbusCallback(): packetId[0x4b], slaveAddress[0xa], functionCode[0x3], numberOfRegisters[22]
[151376][D][main.cpp:417] modbusCallback(): packetId[0x4c], slaveAddress[0xa], functionCode[0x3], numberOfRegisters[44]
[153380][D][main.cpp:417] modbusCallback(): packetId[0x4d], slaveAddress[0xa], functionCode[0x3], numberOfRegisters[22]
[155380][D][main.cpp:417] modbusCallback(): packetId[0x4e], slaveAddress[0xa], functionCode[0x3], numberOfRegisters[44]
[157379][D][main.cpp:417] modbusCallback(): packetId[0x4f], slaveAddress[0xa], functionCode[0x3], numberOfRegisters[22]

**Comment by graham22** [comment]
Reason code 0 is TCP_Disconnected, maybe your wifi is dropping out...
![Capture](https://github.com/ClassicDIY/ClassicMQTT/assets/5341261/a0841526-220b-45f5-afc2-90a54a37f0e8)

**Comment by mcsarge** [comment]
What happens when you run the "Local APP"? That connects directly to your
Classic on port 502. If the computer that is running that is connected by
the same WiFi as the ESP32, and you have some similar problems with
disconnects, the WiFi is the problem.

You could try using the ClassicMQTT on a Raspberry Pi and connect to the
Classic over ethernet.

Matt

On Fri, Feb 2, 2024 at 11:23 AM Graham Ross ***@***.***>
wrote:

> Reason code 0 is TCP_Disconnected, maybe your wifi is dropping out...
> Capture.PNG (view on web)
> <https://github.com/ClassicDIY/ClassicMQTT/assets/5341261/a0841526-220b-45f5-afc2-90a54a37f0e8>
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/ClassicDIY/ClassicMQTT/discussions/41#discussioncomment-8348472>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AABP3HERDE6UDGZ53DCHTMDYRUHH5AVCNFSM6AAAAABCTWFWY2VHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4DGNBYGQ3TE>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***
> com>
>


-- 
Matt Sargent

***@***.***
